app/main.py

```diff
@@ -16,4 +18,22 @@
 }
 collection_of_coins = {1, 2, 25}
 
-# write your code here
+
+def sort_variables(*args) -> dict:
+
+    sorted_dict = {"mutable": [], "immutable": []}
+
+    for current_variable in args:
+        key = "immutable"
+        is_mutable = isinstance(current_variable, (dict, list, set))
+        if is_mutable:
+            key = "mutable"
+
+        sorted_dict[key].append(current_variable)
+
+    return sorted_dict
+
+
+sorted_variables = sort_variables(
+    lucky_number, pi, one_is_a_prime_number, name,
+    my_favourite_films, profile_info, marks, collection_of_coins)
```

Comment:
Here, you’ve made things more complicated. There’s no need to create any functions or variable checks. You just need to manually assign the variables.

Example with other variables:
```python
a = 123
b = []
c = "Hi!"
d = [1, 2]

sorted_variables = {
    "mutable": [b, d],
    "immutable": [a, c]
}
```

Reply:
Yes, I have seen such an implementation from other students. What condition of the task does this implementation not meet?